snuba/state/__init__.py

```diff
@@ -130,33 +143,48 @@ def abtest(value: Optional[Any]) -> Optional[Any]:
     return value
 
 
-def set_config(key: str, value: Optional[Any], user: Optional[str] = None) -> None:
+def set_config(
+    key: str, value: Optional[Any], user: Optional[str] = None, force: bool = False
+) -> None:
     if value is not None:
-        value = "{}".format(value).encode("utf-8")
+        enc_value = "{}".format(value).encode("utf-8")
+    else:
+        enc_value = b""
 
     try:
-        original_value = rds.hget(config_hash, key)
-        if value == original_value:
-            return
+        enc_original_value = rds.hget(config_hash, key)
+        if enc_original_value is not None and value is not None:
+            original_value = numeric(enc_original_value.decode("utf-8"))
+            if value == original_value:
+                return
 
-        change_record = (time.time(), user, original_value, value)
+            if not force and type(value) != type(original_value):
+                raise MismatchedTypeException(type(original_value), type(value))
+
+        change_record = (time.time(), user, enc_original_value, enc_value)
         if value is None:
             rds.hdel(config_hash, key)
-            rds.hdel(config_history_hash, key)
```

Comment:
Removed since we want to maintain history of changes of keys which are deleted as well.